src/core/Akka/Routing/ConsistentHashRouter.cs

```diff
@@ -185,42 +185,43 @@ public override Routee Select(object message, Routee[] routees)
                     return consistentHash;
                 }
                 return oldConsistentHash;
-            };
+            }
 
-            Func<object, Routee> target = hashData =>
+            Routee Target(object hashData)
             {
                 try
                 {
-                    var currentConsistentHash = updateConsistentHash();
+                    var currentConsistentHash = UpdateConsistentHash();
                     if (currentConsistentHash.IsEmpty) return Routee.NoRoutee;
                     else
                     {
-                        if (hashData is byte[])
-                            return currentConsistentHash.NodeFor(hashData as byte[]).Routee;
-                        if (hashData is string)
-                            return currentConsistentHash.NodeFor(hashData as string).Routee;
-                        return
-                            currentConsistentHash.NodeFor(
-                                _system.Serialization.FindSerializerFor(hashData).ToBinary(hashData)).Routee;
+                        switch (hashData)
+                        {
+                            case byte[] bytes:
+                                return currentConsistentHash.NodeFor(bytes).Routee;
+                            case string data:
+                                return currentConsistentHash.NodeFor(data).Routee;
+                            default:
+                                return currentConsistentHash.NodeFor(_system.Serialization.FindSerializerFor(hashData).ToBinary(hashData)).Routee;
+                        }
                     }
                 }
                 catch (Exception ex)
                 {
                     //serialization failed
-                    _log.Value.Warning("Couldn't route message with consistent hash key [{0}] due to [{1}]", hashData,
-                        ex.Message);
+                    _log.Value.Warning("Couldn't route message with consistent hash key [{0}] due to [{1}]", hashData, ex.Message);
                     return Routee.NoRoutee;
                 }
-            };
+            }
 
-            if (_hashMapping(message) != null)
+            var hashMapping = _hashMapping(message);
```

Comment:
This delegate was being called twice which should return the same value each time. So I stored it in a variable so it only needed to be called once.